mods/default/trees.lua

```diff
@@ -426,3 +427,48 @@ function default.grow_new_aspen_tree(pos)
 	minetest.place_schematic({x = pos.x - 2, y = pos.y - 1, z = pos.z - 2},
 		path, "0", nil, false)
 end
+
+
+--
+-- Sapling 'on place' function to check protection of node and resulting tree volume
+--
+
+function default.sapling_on_place(itemstack, placer, pointed_thing,
+		sapling_name, minp_relative, maxp_relative, interval)
+	-- Position of sapling
+	local pos = pointed_thing.under
+	local node = minetest.get_node(pos)
```

Comment:
First of all, that `on_place` function is pretty good!

The only improvement i can think of is handling `CONTENT_IGNORE`, which here is simply done by using:

```lua
local node = minetest.get_node_or_nil(pos)
local pdef = node and minetest.registered_nodes[node.name]
```

It's a rare condition, usually only happening on laggy servers, and even there only directly after teleportation, `/home` etc.
But when it would happen `get_node` would return a `buildable_to` node even without it being `buildable_to`, which would cause `pointed_thing.under` being replaced with a sapling later.
`get_node_or_nil` simply returns `nil` for `CONTENT_IGNORE`